### PR TITLE
Fixes watchdog timer resets (must feed at least once within 8 seconds)

### PR DESCRIPTION
**Pull request by aperepel**
Default example doesn't work on esp8266 (esp12) boards as the loop function doesn't feed the watchdog timer correctly. Figured I'd post this full example for now (I am using PlatformIO-based project structure, not Arduino IDE).

Wondering if including a conditional #ifdef for esp8266 in the PJON::update() function would be an even better fix.

### CODE REVIEW

File: examples/NetworkAnalysis/Transmitter/Transmitter_esp8266 [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// a 10K pull-down resistor is required
+// place it between the GND and the communications pin
+// if you are getting a report of every message in Busy bucket, add a resistor
+
+#if defined(ESP8266)
```
Comment (gioblu) [inline examples/NetworkAnalysis/Transmitter/Transmitter_esp8266]: @aperepel Why you add this condition and inclusion?
With nodeMCU I generally don't have to include ESP8266.h (or is automatically included)

Comment (aperepel) [inline examples/NetworkAnalysis/Transmitter/Transmitter_esp8266]: I think it depends on the toolchain one compiles in, whether it's  included automatically or no. Conditional include, I guess, just defensive programming, so it still compiles if someone tries to put it on e.g. Uno.

**Comment by gioblu** [comment]
@aperepel I agree, if this doesn't create problems with nodeMCU and esp8266 arduino toolchain, will be accepted soon.

**Comment by aperepel** [comment]
One thing still missing is a matching receiver example. The only change is to use the `D1` pin for comms, not 12. I wasn't sure of the approach we'd like to take on providing examples for different boards:

- Use a folder/filename for every architecture?
- Add numerous `#idfef` statements? Not a fan of this either, as it will grow quickly and not too friendly for beginners

**Comment by gioblu** [comment]
@aperepel in my opinion its really a good idea to use the first approach you proposed, that is also how you actually structured this pull-request. For sure will be confortable for the end user and easy to mantain in future.

**Comment by aperepel** [comment]
Sorry I messed up a little with the PR in my fork, but since it's a trivial change I'll post it here. The `Receiver_esp8266.ino` file using the `D1` pin to match the sender:
```
#include <PJON.h>

// network(Arduino pin used, selected device id)
PJON network(D1, 44);

void setup() {
  Serial.begin(115200);
  Serial.println("Starting up");
};

void loop() {

  /*
  receive() function  without parameter
  should not be used because is a low level
  function and is not aware of packets. Here is
  is used only to detect the maximum bandwidth.
  */

  network.receive();
};
```

**Pull request merged**